sidetree/lib/core/interfaces/IDidTypeStore.ts

```diff
@@ -0,0 +1,23 @@
+import DidTypeModel from '../models/DidTypeModel';
+
+/**
+ * An abstraction of a complete store did types.
+ */
+export default interface IDidTypeStore {
+  /**
+   * Inserts or replaces the didtype for a given did.
+   * @param DidTypeModel the did unique suffix and did type
+   */
+  insert (didTypes: DidTypeModel[]): Promise<void>;
+
+  /**
+   * Gets all did unique suffix of the given DID type
+   * @param didType the type of the did
+   */
+  get (didType: string): Promise<string[]>;
```

Comment:
Are we not concerned about a lot of entries in the returned array? Consider future proofing by making small tweaks:
```suggestion
  /**
   * Gets all did unique suffixes of the given DID type that are greater than `greaterThanTransactionNumber` in transaction time ascending order. Fetches all DIDs of the same DID type if `greaterThanTransactionNumber` is `undefined`'. `hasMore` in returned object is set to `true` if there are too many DIDs to return in on call.
   */
  get (didType: string, greaterThanTransactionNumber?: number): Promise<{did: string[], hasMore: boolean}>;
```